maestro-orchestra/src/main/java/maestro/orchestra/Orchestra.kt

```diff
@@ -283,6 +284,10 @@ class Orchestra(
     }
 
     private fun launchAppCommand(it: LaunchAppCommand) {
+        // stop app before launching it, so clear operations have an effect
+        maestro.stopApp(it.appId)
+        MaestroTimer.sleep(MaestroTimer.Reason.BUFFER, 3000)
```

Comment:
We should try to avoid arbitrary sleeps like this - what is the reason for the sleep here?

Reply:
stopApp idb_companion operation finishes almost immediately (same thing for `xcrun simctl terminate`), while it probably takes longer to kill all the simulator app processes, if I remove the delay clearState won't work again.

can experiment with polling a PID of the running app, but I'd say can be done as an improvement.

Reply:
> can experiment with polling a PID of the running app

@artem888 Can you create a task and follow up on this?

Reply:
Sleeps without a validation check at the end make the testing platform flakey.
If it is possible to check the state that's waited on, then we should build that check. (waitloop + exception on timeout). Checking the pid shouldn't be too difficult (and is quite reliable: osses prevent reuse of pids as much as possible).

Reply:
If it's not too difficult I agree with doing the PID check in this change